packages/@jsii/go-runtime-test/project/compliance_test.go

```diff
@@ -104,64 +243,289 @@ func (suite *ComplianceSuite) TestUnionProperties() {
+	}
+}
+
+func (x* overridableProtectedMemberDerived) OverrideReadOnly() string {
+	return "Cthulhu "
+}
+
+func (x* overridableProtectedMemberDerived) OverrideReadeWrite() string {
+	return "Fhtagn!"
+}
+
+func (suite* ComplianceSuite) TestCanOverrideProtectedGetter() {
+	suite.FailTest("Overrides are not supported yet", "https://github.com/aws/jsii/issues/2048")
+
+	assert := suite.Assert()
+	overridden := newOverridableProtectedMemberDerived()
+	assert.Equal("Cthulhu Fhtagn!", overridden.ValueFromProtected())
+}
+
+type implementsAdditionalInterface struct {
+	calc.AllTypes
+	_struct calc.StructB
+}
+
+func (x implementsAdditionalInterface) ReturnStruct() calc.StructB {
+	return x._struct
+}
+
+func newImplementsAdditionalInterface(s calc.StructB) implementsAdditionalInterface {
+	return implementsAdditionalInterface{
+		calc.NewAllTypes(),
+		s,
+	}
+}
+
+func (suite* ComplianceSuite) TestInterfacesCanBeUsedTransparently_WhenAddedToJsiiType() {
+	suite.FailTest("Overrides not supported", "https://github.com/aws/jsii/issues/2048")
+	assert := suite.Assert()
+
+
+	expected := calc.StructB{RequiredString: "It's Britney b**ch!"}
+	delegate := newImplementsAdditionalInterface(expected)
+	consumer := calc.NewConsumePureInterface(delegate)
+	assert.Equal(expected, consumer.WorkItBaby())
+}
+
+func (suite* ComplianceSuite) TestStructs_nonOptionalequals() {
+	assert := suite.Assert()
+
+	structA := calc.StableStruct{ReadonlyProperty: "one"}
+	structB := calc.StableStruct{ReadonlyProperty: "one"}
+	structC := calc.StableStruct{ReadonlyProperty: "two"}
+	assert.Equal(structB, structA)
+	assert.NotEqual(structC, structA)
+}
+
+func (suite* ComplianceSuite) TestTestInterfaceParameter() {
+	assert := suite.Assert()
+
+
+	obj := calc.NewJsObjectLiteralForInterface()
+	friendly := obj.GiveMeFriendly()
+	assert.Equal("I am literally friendly!", friendly.Hello())
+
+	greetingAugmenter := calc.NewGreetingAugmenter()
+	betterGreeting := greetingAugmenter.BetterGreeting(friendly)
+	assert.Equal("I am literally friendly! Let me buy you a drink!", betterGreeting)
+}
+
+func (suite* ComplianceSuite) TestLiftedKwargWithSameNameAsPositionalArg() {
+	assert := suite.Assert()
+
+	// This is a replication of a test that mostly affects languages with keyword arguments (e.g: Python, Ruby, ...)
+	bell := calc.NewBell()
+	amb := calc.NewAmbiguousParameters(bell, calc.StructParameterType{Scope: "Driiiing!"})
+	assert.Equal(bell, amb.Scope())
+
+	expected := calc.StructParameterType{Scope: "Driiiing!"}
+	assert.Equal(expected, amb.Props())
+}
+
+type addTen struct {
+	calc.Add
+}
+
+func newAddTen(value float64) addTen {
+	return addTen{
+		calc.NewAdd(calclib.NewNumber(value), calclib.NewNumber(10)),
+	}
+}
+
+type mulTen struct {
+	calc.Multiply
+}
+
+func newMulTen(value float64) mulTen {
+	return mulTen{
+		calc.NewMultiply(calclib.NewNumber(value), calclib.NewNumber(10)),
+	}
+}
+
+func (suite* ComplianceSuite) TestCreationOfNativeObjectsFromJavaScriptObjects() {
```

Comment:
@RomainMuller what is the right idiom for subclassing a jsii type in go? This approach does not seem to work, but I am not sure if I used the correct pattern.

Reply:
@eladb What approach specifically are you referring to?

Reply:
How do I subclass a jsii class and override methods ?